cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -95,11 +101,18 @@ size_t getContingencyMatrixWorkspaceSize(int nSamples,
 template <typename DataT, typename IdxType>
 size_t getContingencyMatrixWorkspaceSize(const raft::handle_t& handle,
                                          raft::device_vector_view<const DataT, IdxType> groundTruth,
-                                         DataT minLabel = std::numeric_limits<DataT>::max(),
-                                         DataT maxLabel = std::numeric_limits<DataT>::max())
+                                         std::optional<DataT> minLabel = std::nullopt,
```

Comment:
Note that if users provide `nullopt` explicitly as an argument, sometimes the compiler has trouble deducing types. It can be worth adding a test to make sure that compiles. If not, there are a couple different ways to fix it. My PR #830 takes one approach that can handle multiple optional arguments.